hw/virtio/vhost-user.c

```diff
@@ -1648,6 +1717,12 @@ static gboolean slave_read(QIOChannel *ioc, GIOCondition condition,
         ret = vhost_user_slave_handle_vring_host_notifier(dev, &payload.area,
                                                           fd ? fd[0] : -1);
         break;
+    case VHOST_USER_BACKEND_SHARED_OBJECT:
+        ret = vhost_user_backend_handle_shared_object(&payload.object);
```

Comment:
1. Currently, not really a requirement (it is implicitly returning a negative value for the `fd` on errors), but makes the function consistent with other handlers, that also assign `ret` and they may or may not be checked afterward in the ACK response. It could be used if different commands (add or remove) would also require their ACK response. I can discuss that on the `qemu-devel` forums.
2. Yes, it is safe. Error means the operation did not succeed, which for the lookup functions means a negative descriptor will be sent to the backend.